addons/gamification/models/ir_http.py

```diff
@@ -0,0 +1,39 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import time
+from functools import lru_cache
+
+from odoo import models
+from odoo.http import request
+
+EPOCH_DURATION = 300  # 5 min
+
+
+class IrHttp(models.AbstractModel):
+    _inherit = 'ir.http'
+
+    @classmethod
+    def _dispatch(cls, endpoint):
+        result = super()._dispatch(endpoint)
+
+        if uid := request.env.uid:
+            epoch = int(time.time() // EPOCH_DURATION)
```

Comment:
this is not 100% correct

the key
```
int(time.time() // EPOCH_DURATION)
```

and the condition
```
if not user.last_interacted_date or now - user.last_interacted_date > datetime.timedelta(minutes=5):
```

are not 100% equivalent

`00H01` you update `last_interacted_date`
- cache contains a key for `00H00`

`0H05` you try to update it
- cache key become `0H05`
- you call `action_log_interacted_throttled`
- you do not update the value in database (0H05 - 00H01 < 5)

`0H09` you try to update
- the cache key is still `0H05`
- you do not call `action_log_interacted_throttled`


so in total you can have a gap of 10 minutes and not 5, because of this "double logic" stuff, maybe not an issue but it should be considered when we will choose the delay